### PR TITLE
New App Center API

### PR DESCRIPTION
**Pull request by fonkadelic**
This updates the `appcenter*` tasks to the new App Center Upload API.
The implementation is heavily based on https://github.com/microsoft/fastlane-plugin-appcenter/

The IPA and dSYM uploads are now separate tasks in order to provide more flexibility.

### CODE REVIEW

File: plugin/src/test/groovy/org/openbakery/carthage/AbstractCarthageTaskBaseSpecification.groovy [hunk 1]
```diff
@@ -135,6 +136,7 @@ class AbstractCarthageTaskBaseSpecification extends Specification {
 		environment["XCODE_XCCONFIG_FILE"] == xcconfigPath.absolutePath
 	}
 
+	@Ignore
```
Comment (fonkadelic) [inline plugin/src/test/groovy/org/openbakery/carthage/AbstractCarthageTaskBaseSpecification.groovy:139]: I think these tests require some sort of specific environment which is problematic. 
@renep Would be great if you have some feedback here or feel free to push an update here.

File: plugin/src/test/groovy/org/openbakery/carthage/CarthageArchiveTaskSpecification.groovy [hunk 2]
```diff
@@ -157,7 +158,7 @@ class CarthageArchiveTaskSpecification extends Specification {
 	}
 
 
-
+	@Ignore
```
Comment (fonkadelic) [inline plugin/src/test/groovy/org/openbakery/carthage/CarthageArchiveTaskSpecification.groovy:161]: See comment in AbstractCarthageTaskBaseSpecification.groovy

File: plugin/src/test/groovy/org/openbakery/carthage/CarthageBootstrapTaskSpecification.groovy [hunk 3]
```diff
@@ -140,6 +141,7 @@ class CarthageBootstrapTaskSpecification extends Specification {
 	}
 
 
+	@Ignore
```
Comment (fonkadelic) [inline plugin/src/test/groovy/org/openbakery/carthage/CarthageBootstrapTaskSpecification.groovy:144]: See comment in AbstractCarthageTaskBaseSpecification.groovy

File: plugin/src/test/groovy/org/openbakery/carthage/AbstractCarthageTaskBaseSpecification.groovy (hunk 1 above)
Comment (renep) [reply plugin/src/test/groovy/org/openbakery/carthage/AbstractCarthageTaskBaseSpecification.groovy:139]: The tests should not need any special environment. I had the issue that when there was not `/Application/Xcode-12.app` that the unit tests did fail. But I thought I had that fixed. On my box now I only have `/Application/Xcode-12.2.app` and `/Application/Xcode-12.3.app`

I have removed all the `@Ignore` and on my box all unit tests run fine.

**Comment by renep** [comment]
@fonkadelic You can test the develop version `0.21.0.develop.27` that should contain this pull request.

**Pull request merged**